java/lang/security/jackson-unsafe-deserialization.yaml

```diff
@@ -0,0 +1,53 @@
+rules:
+  - id: jackson-unsafe-deserialization
+    patterns:
+      - pattern-either:
+          - patterns:
+              - pattern-inside: |
+                  ObjectMapper $OM = new ObjectMapper(...);

```

Comment:
Recommend multiple `pattern-inside` here - I think you want to ensure that the ObjectMapper is instantiated *and* default typing is enabled, e.g.

```yaml
- patterns:
  - pattern-inside: |
      ObjectMapper $OM = new ObjectMapper(...);
      ...
  - pattern-inside: |
      $OM.enableDefaultTyping();
      ...
  - pattern: $OM.readValue($JSON,...);
```

see https://semgrep.dev/s/PzPv for an example w/test case where object typing is set after an ObjectMapper read

Reply:
That's correct. Thank you for the review!